crates/interpreter/src/host.rs

```diff
@@ -40,6 +40,8 @@ pub trait Host {
     fn effective_gas_price(&self) -> U256;
     /// Transaction caller, calls `ContextTr::tx().caller()`
     fn caller(&self) -> Address;
+    /// Transaction caller as U256, with caching optimization
+    fn caller_u256(&self) -> U256;
```

Comment:
This is not a best abstracton, let us expand LocalContext with a `caller_u256() -> Option<U256>`, so we can check if it is set or not.